### PR TITLE
14133 - Better support for MacOs Alt and Function keys

### PR DESCRIPTION
**Pull request by BrentEaston**
Hi Joel,

I need a snapshot build including this logging change for Mark Benson to do some testing of wtf is actually happening when people use various function and special keys on a Mac.

See http://www.vassalengine.org/forum/viewtopic.php?p=66922#p66922

Thanks.

### CODE REVIEW

**Comment by uckelman** [comment]
Test build: http://www.vassalengine.org/~uckelman/tmp/VASSAL-3.5.4-SNAPSHOT-Macos-Fn-Keys-1fc2696e7-macos.dmg

**Comment by BrentEaston** [comment]
Hi Joel,
Could you make a new test build for Mark with this latest push.
Thanks.

**Comment by uckelman** [comment]
Test build: http://www.vassalengine.org/~uckelman/tmp/VASSAL-3.5.4-SNAPSHOT-HEAD-30c287900-macos.dmg

**Comment by BrentEaston** [comment]
Hi Joel,

This is ready for another test build for Mark.

Thanks,
Brent.

**Comment by uckelman** [comment]
Test build: http://www.vassalengine.org/~uckelman/tmp/VASSAL-3.5.4-SNAPSHOT-HEAD-42ec2e672-macos.dmg

File: vassal-app/src/main/java/VASSAL/build/module/Chatter.java [hunk 1]
```diff
@@ -554,7 +554,7 @@ public void keyCommand(KeyStroke e) {
 
     if ((e.getKeyCode() == 0 || e.getKeyCode() == KeyEvent.CHAR_UNDEFINED)
       && !Character.isISOControl(e.getKeyChar())) {
-      if (((e.getModifiers() & KeyEvent.ALT_DOWN_MASK) != 0) || (e.getModifiers() & KeyEvent.CTRL_DOWN_MASK) != 0)  {
+      if ((e.getModifiers() & KeyEvent.ALT_DOWN_MASK) != 0 || (e.getModifiers() & KeyEvent.CTRL_DOWN_MASK) != 0 || (e.getModifiers() & KeyEvent.ALT_GRAPH_DOWN_MASK) != 0)  {
```
Comment (jonathanrwatts) [inline vassal-app/src/main/java/VASSAL/build/module/Chatter.java:557]: Would it not be faster to do a single compare of e.getModifiers() by or-ing the 3 masks together?  
`if ((e.getModifiers() & (KeyEvent.ALT_DOWN_MASK | KeyEvent.CTRL_DOWN_MASK | KeyEvent.ALT_GRAPH_DOWN_MASK)) != 0)  {`

Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/build/module/Chatter.java]: Not necessarily faster (if Alt key down), but definitely cleaner, which is more important. Thanks! Fixed.

**Comment by BrentEaston** [comment]
Hi Joel,

Could you please merge this into the master and we'll get a new snapshot out to Mark to test everything together. 

Thx.

**Comment by uckelman** [comment]
Test build VASSAL-3.5.4-SNAPSHOT-cbfbe8a9e uploaded:

http://www.vassalengine.org/~uckelman/tmp/

**Pull request merged**